runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java

```diff
@@ -121,27 +122,39 @@ public Dimension render(Graphics2D graphics)
 				}
 			}
 
-			if (config.showIcons())
+			if (config.showSpotTiles())
 			{
-				BufferedImage fishImage = getFishImage(spot);
+				Polygon poly = npc.getCanvasTilePoly();
+				if (poly != null)
+				{
+					OverlayUtil.renderPolygon(graphics, poly, color.darker());
+				}
+			}
+
+			if (config.showSpotIcons())
+			{
+				BufferedImage fishImage = itemManager.getImage(spot.getFishSpriteId());;
 				if (fishImage != null)
 				{
-					OverlayUtil.renderActorOverlayImage(graphics, npc, fishImage, color.darker(), npc.getLogicalHeight());
+					Point imageLocation = npc.getCanvasImageLocation(graphics, fishImage, npc.getLogicalHeight());
+					if (imageLocation != null)
+					{
+						OverlayUtil.renderImageLocation(graphics, imageLocation, fishImage);
+					}
 				}
 			}
-			else
+
+			if (config.showSpotNames())
 			{
 				String text = spot.getName();
-				OverlayUtil.renderActorOverlay(graphics, npc, text, color.darker());
+				Point textLocation = npc.getCanvasTextLocation(graphics, text, npc.getLogicalHeight() + 40);
```

Comment:
Should the height always be incremented by 40? Is this to make space for the icon or for some other reason? (if so, we could check for that and not add the space if the icon is hidden)

Reply:
It is literally copied from the OverlayUtil, that is just where we display text for every actor text overlay

Reply:
Ok :man_shrugging: